src/configure/configure.ts

```diff
@@ -556,17 +556,20 @@ class Orchestrator {
         if (!repoAnalysisResult || !repoAnalysisResult.applicationSettingsList) {
             return;
         }
-        const workingDirectories = Array.from(new Set(this.inputs.potentialTemplates
-            .filter((template) => template.templateType === TemplateType.REMOTE)
-            .map((template: RemotePipelineTemplate) => template.workingDirectory.toLowerCase())));
-        const applicationSettings = repoAnalysisResult.applicationSettingsList.filter((applicationSetting) => {
-            if (this.inputs.pipelineConfiguration.template.templateType === TemplateType.REMOTE) {
-                return workingDirectories.indexOf(applicationSetting.settings.workingDirectory.toLowerCase()) >= 0;
-            }
-            return applicationSetting.language === this.inputs.pipelineConfiguration.template.language;
 
-        });
+        let applicationSettings: ApplicationSettings[] = repoAnalysisResult.applicationSettingsList;
+        if (!this.isResourceAlreadySelected()) {
+            const workingDirectories = Array.from(new Set(this.inputs.potentialTemplates
+                .filter((template) => template.templateType === TemplateType.REMOTE)
```

Comment:
@anuragc617 one question:  should we not be filtering more than just remote templates. We should only be interested in working directories of the templates which match the selected template attributes.